firedrake/cofunction.py

```diff
@@ -31,7 +33,8 @@ class Cofunction(ufl.Cofunction, FunctionMixin):
 
     @PETSc.Log.EventDecorator()
     @FunctionMixin._ad_annotate_init
-    def __init__(self, function_space, val=None, name=None, dtype=ScalarType):
+    def __init__(self, function_space, val=None, name=None, dtype=ScalarType,
+                 count=None):
```

Comment:
The introduction of `count` is essential here due to the `DelegateFunctionCheckpoint` usage when the annotation is activated. Upon recomputing the tape, the [DelegateFunctionCheckpoint.restore](https://github.com/firedrakeproject/firedrake/blob/master/firedrake/adjoint_utils/checkpointing.py#L328) returns a Cofunction with a specified `count`.